src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/gui/action/CorrelationExtractor.java

```diff
@@ -0,0 +1,393 @@
+import org.apache.jmeter.extractor.RegexExtractor;
+import org.apache.jmeter.functions.CorrelationFunction;
+import org.apache.jmeter.gui.GuiPackage;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jmeter.testelement.TestElement;
+import org.apache.jmeter.visualizers.CorrelationRecorder;
+import org.apache.jmeter.visualizers.ViewResultsFullVisualizer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CorrelationExtractor extends ViewResultsFullVisualizer{
+
+    private static final Logger log = LoggerFactory.getLogger(CorrelationExtractor.class);
+
+    // Initialize constants variables
+    static final String CSS_SELECTOR_VARIABLE_NAME = "HtmlExtractor.refname";
+    static final String CSS_SELECTOR_EXPRESSION = "HtmlExtractor.expr";
+    static final String CSS_SELECTOR_ATTRIBUTE = "HtmlExtractor.attribute";
+    static final String CSS_SELECTOR_MATCH_NO = "HtmlExtractor.match_number";
+
+    static final String REGEX_EXTRACTOR_VARIABLE_NAME = "RegexExtractor.refname";
+    static final String REGEX_EXTRACTOR_EXPRESSION = "RegexExtractor.expr";
+    static final String REGEX_EXTRACTOR_TEST_NAME = "testname";
+    static final String REGEX_EXTRACTOR_MATCH_NO = "RegexExtractor.match_number";
+    static final String ONE = "1";
+
+    static final String AMPERSAND = "&";
+    static final String UNDERSCORE = "_";
+    static final String EQUAL = "=";
+    static final String HTTP_SAMPLER_PROXY = "HTTPSamplerProxy";
+    static final String TEST_NAME = "testname";
+    static final String HASHTREE = "<hashTree/>";
+    static final String[] queryParameterRegex = new String[] {
+            "[0-9a-z]*-[0-9a-z]*-[0-9a-z]*-[0-9a-z]*-[0-9a-z]*", "([A-Z])" + "\\" + "w+", "([0-9a-z])\\w+" };
+
+    static final String REGEX = "(.*)[\n]*(.*)";
+    static final String REGEX_EXPRESSION = "(.*?)";
+    static final String ALL_CONTENT_MATCHER = "(.*)";
+    static final String DELIMITER = "|";
+    static final String CONTENTTYPE = "text/html";
+    static final String TEXT_HTML = "text/html";
+    static final String APPLICATION_JSON = "application/json";
+    static List<Map<String, String>> listOfMap = new ArrayList<>();
+
+    private CorrelationExtractor() {}
+
+    /**
+     * Read the response and create the extractor tags.
+     *
+     * @param arguments
+     * @param bodyParameterMap
+     */
+    public static void readResponse(List<String> arguments,
+            Map<String, String> bodyParameterMap) {
+
+        readBufferObject(arguments, bodyParameterMap);
+    }
+
+    /**
+     * Iterate the buffer object and prepare the list of extractor tag.
+     *
+     * @param arguments
+     * @param bodyParameterMap
+     */
+    public static void readBufferObject(List<String> arguments,
+            Map<String, String> bodyParameterMap) {
+
+        for (Object sampler : CorrelationRecorder.buffer) {
+            log.info("processing of buffer objects start.");
```

Comment:
I think `info` is too high a level for this, especially inside a loop, perhaps `debug`?